modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/fragment/renderer/FDSViewFragmentRenderer.java

```diff
@@ -454,7 +454,17 @@ private JSONArray _getFiltersJSONArray(
 					).put(
 						"id", properties.get("fieldName")
 					).put(
-						"label", properties.get("name")
+						"label",
+						() -> {
+							String label = String.valueOf(
```

Comment:
this seems candidate for a separate function that returns the filter label. We're using it twice, but we'll end up using it 3 times thanks to the ClientExtension filter PR, yet to come